.github/workflows/run-tests-and-mypy.yml

```diff
@@ -0,0 +1,41 @@
+name: Run tests and mypy
+
+on:
+  push:
+      branches: [ dev ]
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
```

Comment:
Is this standard ? (not super familiar with github actions) Is this not going to trigger redundant pipeline, or is github able to spot them ?

Reply:
It depends on what you behavior we want, this will run on merges to dev, and when opening PRs, when pushing new commits to PRs, when reopening closed PRs - all assuming the PR was not marked as draft
and when switching a PR from marked as draft to ready for review.

It's how we had it in Qu & Co

Reply:
cool cool